File: 03-1-login_kt2/app/src/main/java/com/example/login_kt2/repository/LoginTryRepository.kt
package com.example.login_kt2.repository

import com.example.login_kt2.repository.room.LoginTry
import com.example.login_kt2.repository.room.LoginTryDao
import kotlinx.coroutines.flow.Flow

class LoginTryRepository(private val dao: LoginTryDao) {

    suspend fun insert(loginTry: LoginTry) = dao.insert(loginTry)

    fun getAll() : Flow<List<LoginTry>> = dao.getAllInFlow()

}

File: 03-1-login_fl/lib/repository/login_try_repository.dart
import 'package:login_fl/repository/floor/login_try.dart';
import 'package:login_fl/repository/floor/login_try_dao.dart';

class LoginTryRepository {

  final LoginTryDao loginTryDao;

  LoginTryRepository(this.loginTryDao);

  Future<void> insert(LoginTry loginTry) => loginTryDao.insert(loginTry);

  Future<List<LoginTry>> getAll() => loginTryDao.getAll();

  Stream<List<LoginTry>> getAllInStream() => loginTryDao.getAllInStream();
}